### DISCUSSION TITLE
Opentelemetry alpha versions

### DISCUSSION THREAD

**Discussion by mederel** [quarkusio/quarkus#28173] [Q&A] [answered]
I recently discovered that some of the dependencies from the Opentelemetry extension have versions suffixed with -alpha. Notably instrumentation and semconv. Is it not a concern to depend on alpha versions of components?

[a comment by an automated account is not shown]

**Reply by brunobat** [reply]
Hi @mederel, adding to Roberto.
The protocol, SDK and APIs for tracing are stable. Things like semantic conventions are still under development but there is no impediment to using them.
Example: io.opentelemetry.semconv.resource.attributes.ResourceAttributes
These are generated from the spec and are still evolving. Worst case, you will need to update constants in the code or your queries

**Reply by mederel** [reply]
> Hi @mederel, adding to Roberto. The protocol, SDK and APIs for tracing are stable. Things like semantic conventions are still under development but there is no impediment to using them. Example: io.opentelemetry.semconv.resource.attributes.ResourceAttributes These are generated from the spec and are still evolving. Worst case, you will need to update constants in the code or your queries

I'd add you also as an answer but there can be only one to rule them all :cry:

**Comment by radcortez** [comment, marked as answer]
I understand your concern.

On the other hand, when we consider what the alpha dependencies bring to the table, we believe they are worth using. Also, we are involved with the OpenTelemetry community, and we are confident that these alpha dependencies will end up with final versions.
